### PR TITLE
[ATL][MSPAINT] Add ATLTRACE and fix assertion failures

### PR DESCRIPTION
**Pull request by katahiromz**
## Purpose
Improve traceability of ATL applications.
JIRA issue: [CORE-17969](https://jira.reactos.org/browse/CORE-17969), [CORE-18012](https://jira.reactos.org/browse/CORE-18012)

## Proposed changes

- Implement `ATLTRACE` and `AtlTrace` in `atlbase.h`.
- Fix assertion failures in `mspaint`.

## Screenshot
無題

### CODE REVIEW

File: sdk/lib/atl/atlbase.h [hunk 1]
```diff
@@ -1856,6 +1859,61 @@ inline HRESULT WINAPI AtlComModuleRevokeClassObjects(_ATL_COM_MODULE *module)
     return S_OK;
 }
 
+#if DBG
```
Comment (HBelusca) [inline sdk/lib/atl/atlbase.h:1862]: ATL Tracing functionality is defined in `atltrace.h`

Comment (katahiromz) [reply sdk/lib/atl/atlbase.h]: OK. Moved.

File: sdk/lib/atl/atltrace.h [hunk 2]
```diff
@@ -0,0 +1,203 @@
...
+#include <stdio.h>
+
+namespace ATL
+{
+
+struct CTraceCategory
+{
+    explicit CTraceCategory(LPCSTR name, UINT level = 0) { }
+
+    operator UINT() const { return (1 << 15); }
```
Comment (learn-more) [inline sdk/lib/atl/atltrace.h:32]: Shouldn't this be the category instead of a hardcoded number?

File: sdk/lib/atl/atltrace.h [hunk 3]
```diff
@@ -0,0 +1,203 @@
...
+    static UINT GetLevel()      { return s_level; }
+    static UINT GetCategories() { return s_categories; }
+    static void SetLevel(UINT level)           { s_level = level; }
+    static void SetCategories(UINT categories) { s_categories = categories; }
+
+    static bool IsTracingEnabled(UINT category, UINT level)
+    {
+        if (s_level == DisableTracing || s_level < level || !(s_categories & category))
+            return false;
+        return ::IsDebuggerPresent();
```
Comment (learn-more) [inline sdk/lib/atl/atltrace.h:69]: I don;'t think this should depend on the presence of a usermode debugger?

File: sdk/lib/atl/atltrace.h [hunk 4]
```diff
@@ -0,0 +1,203 @@
...
+#ifdef _STRSAFE_H_INCLUDED_
+    StringCchPrintfW(szBuff, _countof(szBuff), L"%hs(%d) : ", file, line);
+    StringCchLengthW(szBuff, _countof(szBuff), &cch);
+    StringCchVPrintfW(&szBuff[cch], _countof(szBuff) - cch, format, va);
+#else
+    cch = _snwprintf(szBuff, _countof(szBuff), L"%hs(%d) : ", file, line);
+    _vsnwprintf(&szBuff[cch], _countof(szBuff) - cch, format, va);
+#endif
+
+    OutputDebugStringW(szBuff);
```
Comment (learn-more) [inline sdk/lib/atl/atltrace.h:129]: If you use `_CrtDbgReportW` there should be enough customization options?

File: sdk/lib/atl/atltrace.h (hunk 3 above)
Comment (katahiromz) [reply sdk/lib/atl/atltrace.h]: Deleted `::IsDebuggerPresent();` call.

File: sdk/lib/atl/atltrace.h (hunk 4 above)
Comment (katahiromz) [reply sdk/lib/atl/atltrace.h]: OK.

File: sdk/lib/atl/atltrace.h (hunk 2 above)
Comment (katahiromz) [reply sdk/lib/atl/atltrace.h]: OK. Fixed.

**Comment by katahiromz** [comment]
Should be OK now.

File: base/applications/mspaint/winproc.cpp [hunk 5]
```diff
@@ -73,13 +73,18 @@ void CMainWindow::alignChildrenToMainWindow()
     }
 
     RECT statusBarRect0;
```
Comment (tkreuzer) [inline base/applications/mspaint/winproc.cpp:75]: This variable is now potentially uninitialized.

Comment (tkreuzer) [reply base/applications/mspaint/winproc.cpp:75]: And with "potentially" I mean for sure.

**Pull request merged**